src/basic/process-util.c

```diff
@@ -1727,6 +1728,41 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+        assert_se(sigaddset(&mask, SIGCHLD) >= 0);
+
+        r = posix_spawnattr_init(&attr);
+        if (r < 0)
+                return r;
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK);
+        if (r < 0)
+                goto out;
+        r = posix_spawnattr_setsigmask(&attr, &mask);
+        if (r < 0)
+                goto out;
+
+        r = posix_spawn(&pid, path, NULL, &attr, argv, envp);
+        if (r < 0)
+                goto out;
+
+        *ret_pid = pid;
+
+out:
+        posix_spawnattr_destroy(&attr);
```

Comment:
Might be overkill, but I'd like to use `_cleanup_` attribute for the finalization.

Reply:
yeah, that'd be cool

Reply:
no addressed.

Reply:
this broke a bunch of stuff - see later comments

Reply:
where?

Reply:
Somewhere in one of the ~400 hidden comments. IIRC the compiler/static analyzer wasn't happy because it's not a pointer so it can't be initialized to NULL so the destroy can run on uninitialzed var in case the initialization itself fails, or something like that. I wasted an hour on it and that was more than enough, given it's purely cosmetic.